src/Entities/Hand.ts

```diff
@@ -1,20 +1,60 @@
-import Card from "./Card";
+import Card from './Card'
 
 class Hand {
-  private hand: Card[];
+  private hand: Card[]
 
   constructor() {
-    this.hand = [];
+    this.hand = []
   }
 
-  public getCard(cardId: string): Card {
-    // find the card that has that cardId and then remove and return it
-    throw Error("get card not implemented");
+  public addCard(card: Card): void {
+    this.hand.push(card)
   }
 
-  public addCard(card: Card): void {
-    this.hand.push(card);
+  public removeCardFromHand(cardId: string): Card {
+    if (this.hand.some((card) => card.getCardId() === cardId)) {
+      const result = this.hand.find((card) => card.getCardId() === cardId)
+      this.hand = this.hand.filter((card) => card.getCardId() !== cardId)
+      return result
+    } else {
+      throw Error(`Card id: ${cardId} not in hand`)
+    }
+  }
+
+  public getPlayableCardIds(leadCard?: Card): string[] {
+    if (leadCard === undefined) {
+      return this.getAllCardIds()
+    }
+    if (this.mustPlayTrump(leadCard)) {
+      return this.getTrumpIds()
+    }
+    if (this.mustPlayMatchingSuit(leadCard)) {
+      return this.getNonTrumpMatchingSuitIds(leadCard)
+    }
+    return this.getAllCardIds()
+  }
+
+  private mustPlayTrump(leadCard: Card): boolean {
+    return leadCard.isTrump() && this.hand.some((card) => card.isTrump())
+  }
+
+  private getTrumpIds(): string[] {
+    return this.hand.filter((card) => card.isTrump()).map((card) => card.getCardId())
+  }
+
+  private mustPlayMatchingSuit(leadCard: Card): boolean {
+    return this.hand.some((card) => !card.isTrump() && card.getSuit() === leadCard.getSuit())
+  }
+
+  private getNonTrumpMatchingSuitIds(leadCard: Card): string[] {
+    return this.hand
+      .filter((card) => !card.isTrump() && card.getSuit() === leadCard.getSuit())
+      .map((card) => card.getCardId())
```

Comment:
there are a few functions on arrays that I don't think python does exactly the same way.

You may be understanding them without any trouble but I figured I would talk about them.

`some` returns true if the callback function returns true for at least one element
`find` returns the first element that the callback function returns true for
`map` returns a new array from the old array by turning each element from the first array into a new element in the new array
`filter` returns an array containing only the elements for which the callback function returned true